Modules/Core/Common/include/itkMetaDataObject.h

```diff
@@ -149,6 +149,52 @@ class ITK_TEMPLATE_EXPORT MetaDataObject : public MetaDataObjectBase
   ~MetaDataObject() override = default;
 
 private:
+  /** Assigns the value of `source` to `target`.
+   * \note The trailing return type is there, just to enable SFINAE.*/
+  template <typename TValue>
+  static auto
+  Assign(TValue & target, const TValue & source) -> decltype(target = source)
+  {
+    return target = source;
+  }
+
+  /** `Assign` overload for C-style arrays (as well as arrays of arrays). */
+  template <typename TValue, size_t VNumberOfElements>
+  static void
```

Comment:
Should / can this return a reference to the left hand side?

Reply:
@Leengit Thanks for asking. The `Assign` overload for C-style arrays might indeed return a reference to the left hand side, but I would not say it _should_. Because it isn't really necessary. Note that it is just an internal private member function (template). Technically, it is not a template specialization of the other `Assign(target, source)`, it is just an overload. (If it wound have been a template specialization, it would be required to have the same return type as the generic function template. But that's not the case here.)